include/real/real_data.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef BOOST_REAL_REAL_DATA_HPP
+#define BOOST_REAL_REAL_DATA_HPP
+
+#include <variant>
+#include <assert.h>
+#include <iostream>
+
+#include "real/const_precision_iterator.hpp"
+#include "real/interval.hpp"
+#include "real/real_explicit.hpp"
+#include "real/real_algorithm.hpp"
+#include "real/real_operation.hpp"
+
+namespace boost { 
+    namespace real{
+        typedef std::variant<std::monostate, real_explicit, real_algorithm, real_operation> real_number;
+
+        class real_data {
+            real_number _real;
+            const_precision_iterator _precision_itr;
```

Comment:
Why the data knows the iterator?

Reply:
A real contains its iterator, and everything a real contains is held in real_data.